src/Utilities/Math.hpp

```diff
@@ -19,3 +22,22 @@ SPECTRE_ALWAYS_INLINE T number_of_digits(const T number) {
   return number == 0 ? 1 : static_cast<decltype(number)>(
                                std::ceil(std::log10(std::abs(number) + 1)));
 }
+
+/*!
+ * \ingroup UtilitiesGroup
+ * \brief Evaluate a polynomial with Horner's rule
+ *
+ * \tparam U The type of the polynomial coefficients \p coeffs. Can be `double`,
+ * which means the coefficients are constant for all values in \p x. Can also be
+ * a vector type of typically the same size as `T`, which means the coefficients
+ * vary with the elements in \p x.
+ * \tparam T The type of the polynomial variable \p x. Must support
+ * `make_with_value<T, T>`, as well as (elementwise) addition with `U` and
+ * multiplication with `T`.
```

Comment:
Document which order the coefficients are in (constant to largest power or vice versa).